### PR TITLE
Scheduler, Notification, Multi Connector check run support

### PR DESCRIPTION
**Pull request by jayasimha-raghavan-unskript**
## Description

[EN-5111]

Adds Support for
* CRUD support for Notification 
* Multi Connector Checks Run support
* Write up on how to add Scheduler to run the Checks periodically. 

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

#### Notification Create
https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/2c014d43-2a5e-45a2-8048-fdc55f82d9e8

#### Notification Read

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/7698de4f-d4da-4c61-bb9b-1750081e4b6d

#### Notification Delete

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/11226683-ef38-46d5-a016-691d116f7557

#### Notification Update + Multiselection + Slack Notification

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/0eaacacc-b8c7-4dec-994b-fe8888230f53

#### Notification Updtate + Multiselection + Email Notification

https://github.com/unskript/Awesome-CloudOps-Automation/assets/87547684/77ade7a5-8026-445c-89a9-88124065d050

### Checklist:

- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository. 

[EN-5111]: https://unskript.atlassian.net/browse/EN-5111?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: README_extending_docker.md [hunk 1]
```diff
@@ -175,3 +175,99 @@ of that file in `unskript-ctl` Folder.
...
+add_notification.sh -c Slack -u https://hooks.slack.com/services/T12345/B12345/XXXXXXX --channel test-alerting
+```
+> Here `-c` Option is used for creating a new Notification. Options are either Slack or SMTP (email)
+> `-u` Option is the webhook URL of Slack 
+> `--channel` Channel to where the notification should be sent
+
+This snippet shows how to configure SMTP (email) Notification
+
+```
+./add_notification.sh -c SMTP -s smtp.server.com -u username@domain.com -p <password_of_username> -t reciever@example.com
```
Comment (jayasimha-raghavan-unskript) [inline README_extending_docker.md:259]: Spell mistake receiver

File: README_extending_docker.md [hunk 2]
```diff
@@ -175,3 +175,99 @@ of that file in `unskript-ctl` Folder.
...
+> Here `-c` Option is used for creating a new Notification. Options are either Slack or SMTP (email)
+> `-u` Option is the webhook URL of Slack 
+> `--channel` Channel to where the notification should be sent
+
+This snippet shows how to configure SMTP (email) Notification
+
+```
+./add_notification.sh -c SMTP -s smtp.server.com -u username@domain.com -p <password_of_username> -t receiver@example.com
+```
+> Here `-s` Option is to specifiy the SMTP server
```
Comment (amit-chandak-unskript) [inline README_extending_docker.md:261]: Please run spell checker on your changes, specify is misspelled everywhere

File: README_extending_docker.md [hunk 3]
```diff
@@ -175,3 +175,99 @@ of that file in `unskript-ctl` Folder.
...
+
+This snippet shows how to configure SMTP (email) Notification
+
+```
+./add_notification.sh -c SMTP -s smtp.server.com -u username@domain.com -p <password_of_username> -t receiver@example.com
+```
+> Here `-s` Option is to specifiy the SMTP server 
+> `-u` Option is to specify the SMTP username. Note, you need to specificy the username with the domain like username@domain.com 
+> `-p` Option is to specify the Password for the above user. Please note, if you are using Gmail as SMTP server, then make sure you create a App Password and use that for the `-p` option
+> `-t` Optino is to specify the Reciever's Email address to whom the Email should be sent to
```
Comment (amit-chandak-unskript) [inline README_extending_docker.md:264]: Option, receiver spelling

File: README_extending_docker.md [hunk 4]
```diff
@@ -175,3 +175,99 @@ of that file in `unskript-ctl` Folder.
...
+> `--channel` Channel to where the notification should be sent
+
+This snippet shows how to configure SMTP (email) Notification
+
+```
+./add_notification.sh -c SMTP -s smtp.server.com -u username@domain.com -p <password_of_username> -t receiver@example.com
+```
+> Here `-s` Option is to specifiy the SMTP server 
+> `-u` Option is to specify the SMTP username. Note, you need to specificy the username with the domain like username@domain.com 
+> `-p` Option is to specify the Password for the above user. Please note, if you are using Gmail as SMTP server, then make sure you create a App Password and use that for the `-p` option
```
Comment (amit-chandak-unskript) [inline README_extending_docker.md:263]: an App password, can we put like some hyperlink?

File: README_extending_docker.md (hunk 3 above)
Comment (amit-chandak-unskript) [reply README_extending_docker.md:264]: why is Email in capital E?

File: README_extending_docker.md [hunk 5]
```diff
@@ -175,3 +175,99 @@ of that file in `unskript-ctl` Folder.
...
+> `-t` Optino is to specify the Reciever's Email address to whom the Email should be sent to
+
+
+4. You are not set to receive the report whenever you run the check with the `--report` option
```
Comment (amit-chandak-unskript) [inline README_extending_docker.md:267]: you are not set???? Could you please go over your README.md thoroughly to rule out basic stuff, spelling mistake, capitalized words n all

File: unskript-ctl/add_notification.py [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+            return 
+        if type.lower() == 'slack':
+            self.SLACK()
+        elif type.lower() == 'smtp':
+            self.SMTP()
+        else:
+            print("ERROR: Wrong choice")
+
+    def read_notification(self, type):
+        """read_notification: Implementing cuRd for Notification
```
Comment (amit-chandak-unskript) [inline unskript-ctl/add_notification.py:154]: This is only implementing read, not curd?? (crud)

File: unskript-ctl/unskript-client.py [hunk 7]
```diff
@@ -403,7 +413,18 @@ def run_checks(args: list):
...
     elif filter == '--type':
-            check_list = get_checks_by_connector(args[-1], True)
+        check_list = []
+        all_connectors = args[1:]
+        if args[-1] in ('-r', '--report'):
+            UNSKRIPT_GLOBALS['report'] = True
+            all_connectors = args[1:-1]
+            
+        for connector in all_connectors:
+            connector = connector.replace(',', '')
```
Comment (amit-chandak-unskript) [inline unskript-ctl/unskript-client.py:423]: What if you give types without a space like
```
--type aws,k8s,jira
```
You should handle this as well.

File: unskript-ctl/add_notification.py (hunk 6 above)
Comment (jayasimha-raghavan-unskript) [reply unskript-ctl/add_notification.py:154]: :) Spelling and Me!

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**